### PR TITLE
Introduce typings for event listeners

### PR DESCRIPTION
**Pull request by weili-govtech**
**Changes**
- Introduce typings for `addFieldEventListener`, `dispatchFieldEvent` and `removeFieldEventListener`
- This is done via event overloading
- Note: This is just syntactic sugar to ease usage of the event listener methods and these new signatures are not used internally

```
// old way (still supported)
formRef.current.addFieldEventListener("click", "fieldId", myEventListener);
formRef.current.dispatchFieldEvent("update-image-status", "fieldId", myPayload);
formRef.current.removeFieldEventListener("click", "fieldId", myEventListener);

// new way with uiType as first argument
formRef.current.addFieldEventListener("button", "click", "fieldId", myEventListener);
formRef.current.dispatchFieldEvent("image-upload", "update-image-status", "fieldId", myPayload);
formRef.current.removeFieldEventListener("button", "click", "fieldId", myEventListener);
```

### CODE REVIEW

File: src/components/fields/location-field/types.ts
Comment (qroll) [inline src/components/fields/location-field/types.ts:1]: there seems to be a field event for `set-current-location` (which is different from the trigger?)

**Review by qroll — COMMENTED** [review]
having autocompletion functionality is great 👍

File: src/components/fields/location-field/types.ts
Comment (weili-govtech) [reply src/components/fields/location-field/types.ts:1]: yeah this implementation is rather confusing, it is a mean for various components to communicate with each other. the component needs major refactoring in the future though I am not sure when we can make it happen

Comment (qroll) [reply src/components/fields/location-field/types.ts:1]: meaning `set-current-location` event is internal and doesn't need to be included in the public typings?

Comment (weili-govtech) [reply src/components/fields/location-field/types.ts:1]: yes it's internal and not meant to be called externally, hence not exposed in the public typings

**Pull request merged**